modules/integration/tests-integration/tests-service/src/test/java/org/wso2/carbon/esb/rest/test/header/ESBJAVA2283ReturnContentTypeTestCase.java

```diff
@@ -18,52 +18,80 @@
-        super.init();
-        loadESBConfigurationFromClasspath("/artifacts/ESB/synapseconfig/esbjava2283/synapse.xml");
+	    super.init();
+	    String relativePath = "/artifacts/ESB/synapseconfig/esbjava2283/api.xml";
+	    ESBTestCaseUtils util = new ESBTestCaseUtils();
+	    relativePath = relativePath.replaceAll("[\\\\/]", File.separator);
+	    OMElement apiConfig = util.loadResource(relativePath);
+	    addApi(apiConfig);
     }
 
     @Test(groups = {"wso2.esb"}, description = "test return content type")
     public void testReturnContentType() throws Exception {
-        SimpleWebServer simpleWebServer = new SimpleWebServer(9005, 200);
-        try {
-            simpleWebServer.start();
-            SimpleHttpClient httpClient = new SimpleHttpClient();
-            Map<String, String> headers = new HashMap<String, String>();
-            headers.put("Content-Type", "text/xml");
-            HttpResponse response = httpClient.doGet(context.getContextUrls().getServiceUrl() + "/passThoughProxy", headers);
-            log.info(response.getEntity().getContentType());
-            log.info(response.getStatusLine().getStatusCode());
+	    int port = 8089;
+	    String contentType = "text/xml";
+	    server = HttpServer.create(new InetSocketAddress(port), 0);
+	    server.createContext("/gettest", new MyHandler());
+	    server.setExecutor(null); // creates a default executor
+	    server.start();
+	    DefaultHttpClient httpclient = new DefaultHttpClient();
+	    String url = "http://localhost:8280/serviceTest/test";
+	    HttpGet httpGet = new HttpGet(url);
+	    HttpResponse response = null;
+	    try {
+		    response = httpclient.execute(httpGet);
+	    } catch (IOException e) {
+		    log.error("Error Occurred while sending http get request. " + e);
+	    }
+	    log.info(response.getEntity().getContentType());
+	    log.info(response.getStatusLine().getStatusCode());
+
+	    assertEquals(response.getFirstHeader("Content-Type").getValue(), contentType,
+	                 "Expected content type doesn't match");
+	    assertEquals(response.getStatusLine().getStatusCode(), 200, "response code doesn't match");
 
-            assertEquals(response.getEntity().getContentType().getValue(), "text/xml", "Expected content type doesn't match");
-            assertEquals(response.getStatusLine().getStatusCode(), 200, "response code doesn't match");
-        } finally {
-            simpleWebServer.terminate();
-            ClientConnectionUtil.waitForPort(9005, InetAddress.getLocalHost().getHostName());
-        }
+	    server.stop(0);
     }
 
+	private class MyHandler implements HttpHandler {
+		public void handle(HttpExchange t) throws IOException {
+			Headers h = t.getResponseHeaders();
+			h.add("Content-Type", "text/xml");
+			String response = "This is the test case for ESBJAVA-2283";
+			t.sendResponseHeaders(200, response.length());
+			OutputStream os = t.getResponseBody();
+			os.write(response.getBytes());
+			os.close();
```

Comment:
Output stream is not closed in a finally block, which can lead to potential memory leaks..